src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -356,6 +358,32 @@ private GetMappingsRequest buildGetMappingRequest(String indexName) {
         return getMappingsRequest;
     }
 
+    private static void validatePPLToolParameters(Map<String, Object> map) {
+        if (!map.containsKey("model_id")) {
+            throw new IllegalArgumentException("PPL tool needs model id.");
+        } else {
+            String modelId = (String) map.get("model_id");
+            if (StringUtils.isBlank(modelId)) {
+                throw new IllegalArgumentException("PPL tool needs non blank model id.");
+            }
+        }
+        if (map.containsKey("execute")) {
+            String execute = map.get("execute").toString().toLowerCase(Locale.ROOT);
+            if (!(execute.equals("true")) && !(execute.equals("false"))) {
+                throw new IllegalArgumentException("PPL tool parameter execute must be false or true");
+            }
+
+        }
+        if (map.containsKey("head")) {
+            String head = map.get("head").toString();
+            try {
+                int headInt = NumberUtils.createInteger(head);
```

Comment:
Do we need to check if the head is positive?

Reply:
I change the logic to prevent it. Please check it. From my perspective, these parameters are from register body and couldn't be null. But anyway, it's better to have it.